src/DialogflowEntityClient.js

```diff
@@ -0,0 +1,52 @@
+import { EntityTypesClient } from 'dialogflow';
+
+export class EntityNotFoundError extends Error { }
+
+export class DialogflowEntityClient {
+  constructor(entityTypeName, projectId) {
+    this.entityTypeName = entityTypeName;
+    this.projectId = projectId;
+  }
+
+  async initialize() {
+    if (!this.entitiesClient) {
+      this.entitiesClient = await new EntityTypesClient();
+    }
+    if (!this.agentPath) {
+      this.agentPath = await this.entitiesClient.projectAgentPath(this.projectId);
+    }
+  }
+
+  async entityTypesList() {
+    return this.entitiesClient.listEntityTypes({ parent: this.agentPath });
+  }
+
+  async entityTypeObject() {
+    const entityTypesList = await this.entityTypesList();
+    const entity = entityTypesList[0]
+      .find(resource => resource.displayName === this.entityTypeName);
+    if (entity) {
+      return entity;
+    }
+    throw EntityNotFoundError;
```

Comment:
Please do if(!entity) { throw EntityNotFoundError }.
Side effects should not be the last statement.